module-1/lab-mysql-select/your-code/solutions.sql

```diff
@@ -0,0 +1,45 @@
+#challenge 1
+
+SELECT authors.au_id AS 'AUTHOR ID', authors.au_lname AS 'LAST NAME', authors.au_fname AS 'FIRST NAME', titles.title AS TITLE, publishers.pub_name AS PUBLISHER
+FROM authors, titles, publishers
+WHERE title_id = title_id
+
+#challenge 2
+
+SELECT 	authors.au_id AS 'AUTHOR ID', 
+		authors.au_lname AS 'LAST NAME', 
+        authors.au_fname AS 'FIRST NAME', 
+        publishers.pub_name AS 'PUBLISHER',
+        COUNT(titles.title_id) AS 'TITLE COUNT'
+FROM authors
+LEFT JOIN titleauthor
+ON titleauthor.au_id=authors.au_id
+LEFT JOIN titles
+ON titles.title_id=titleauthor.title_id 
+LEFT JOIN publishers
+ON titles.pub_id=publishers.pub_id
+GROUP BY authors.au_id, publishers.pub_name;
+
+#challenge 3
+
+SELECT authors.au_id AS 'AUTHOR ID', authors.au_lname AS 'LAST NAME', authors.au_fname AS 'FIRST NAME', SUM(sales.qty) AS 'TOTAL'
+FROM authors
+INNER JOIN titleauthor
+ON titleauthor.au_id=authors.au_id
+INNER JOIN sales
+ON titleauthor.title_id=sales.title_id
+GROUP BY authors.au_id
+ORDER BY SUM(sales.qty) DESC
+LIMIT 3 ;
+
+#challenge 4
+
+SELECT authors.au_id AS 'AUTHOR ID', authors.au_lname AS 'LAST NAME', authors.au_fname AS 'FIRST NAME', SUM(sales.qty) AS 'TOTAL'
+FROM authors
+LEFT JOIN titleauthor
+ON titleauthor.au_id=authors.au_id
+LEFT JOIN sales
+ON titleauthor.title_id=sales.title_id
+GROUP BY authors.au_id
+ORDER BY SUM(sales.qty) DESC
+
```

Comment:
En esta challenge habia que convertir el `null` in uno 0, para hacer eso una solucción puede ser esta: `COALESCE(sum(s.qty), 0) as TOTAL` el coalesce devuelve el primer null en la lista y lo puedes cambiar por otro valor